facade/src/main/java/org/jboss/pnc/facade/mapper/api/BuildMapper.java

```diff
@@ -101,6 +109,21 @@ default BuildRecord toIDEntity(BuildRef dtoEntity) {
     @BeanMapping(ignoreUnmappedSourceProperties = {"project", "repository"})
     BuildRecord toEntity(Build dtoEntity);
 
+
+
+    @Mapping(target = "project", source = "buildConfigurationAudited.project", resultType = ProjectRef.class)
+    @Mapping(target = "repository", source = "buildConfigurationAudited.repositoryConfiguration")
```

Comment:
```suggestion
    @Mapping(target = "repository", source = "buildConfigurationAudited.repositoryConfiguration", qualifiedBy = Reference.class)
```
without Reference.class mapstruct uses toDTO instead of toRef method for mapping the SCMRepository, it can cause confusion and it's good practice to keep the mappers clean :)